### ISSUE TITLE
False Negative when referencing missing members on known modules with --ignore-missing-imports

### ISSUE THREAD

**Issue by mgilson** [python/mypy#7428]
With the advent of mypy==0.720, it appears that we have lost the ability to detect when members are missing from a module if we are using `--ignore-missing-imports`.

Consider the following python program:

```py
import json
json.not_a_member  # Clearly a bug.
```

The following issues an error:

```
$ mypy --strict ~/sandbox/foo.py
/home/mgilson/sandbox/foo.py:2: error: Module has no attribute "not_a_member"
```

However:

```
$ mypy --strict --ignore-missing-imports ~/sandbox/foo.py
```

does not complain.

If you run that with `mypy==0.700` it does complain (I didn't check mypy==0.710), but I assume that it will complain as well.

This seems to be controlled by the semantic analyzer version:

```
$ mypy --strict --ignore-missing-imports --no-new-semantic-analyzer ~/sandbox/foo.py
/home/mgilson/sandbox/foo.py:2: error: Module has no attribute "not_a_member"
```

**Comment by mgilson** [comment]
I should also mention that I can get around this if I explicitly whitelist things in my `mypy.ini`:


```
[mypy]

[mypy-unstubbed_third_party]
ignore_missing_imports = True
```

And then run:

```
mypy --strict program_that_uses_unstubbed_third_party.py
```

But that is pretty cumbersome when running on an existing project that includes more than a handful of third-party dependencies.

**Comment by JukkaL** [comment]
This seems to be fixed on mypy git master. Can you give it a try?

**Comment by mgilson** [comment]
Yep, I can confirm that this is fixed in the master branch.  I tried briefly to bisect to the point where it got fixed, but my git-fu wasn't quite strong enough for me to get it to work without real effort (it kept complaining about the commits that I was picking not being in the ancestry tree or something) and I didn't have a ton of time to spend digging through the history to figure it out.

**Comment by ilevkivskyi** [comment]
@mgilson No need to look for it, I confess, I did it in  https://github.com/python/mypy/pull/7372

**Comment by mgilson** [comment]
ah ha!

And you wrote a regression test which was the whole reason I was looking in the first place.  Nice (and thank you!)
